### PR TITLE
feat: add jemalloc feature and fix memory leak

### PR DESCRIPTION
**Pull request by KaoImin**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

### CI Switch

ci-runs-only: [Chaos CI, Cargo Clippy, Coverage Test, E2E Tests, Code Format, Unit Tests]

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -50,11 +41,13 @@ members = [
...
 default = []
-random_leader = ["core-consensus/random_leader"]
+jemalloc = ["core-run/jemalloc"]
 pprof = ["core-api/pprof"]
+random_leader = ["core-consensus/random_leader"]
 # tentacle_metrics = ["core-network/tentacle_metrics"]
+
+[profile.release]
+overflow-checks = true
+debug = true
```
Comment (driftluo) [inline Cargo.toml]: remove it by default

File: core/run/Cargo.toml [hunk 2]
```diff
@@ -9,6 +9,9 @@ backtrace = "0.3"
 cita_trie = "3.0"
 clap = "3.1"
 futures = "0.3"
+jemalloc-ctl = { version = "0.4", package = "tikv-jemalloc-ctl", optional = true }
+jemallocator = { version = "0.4", features = ["profiling", "stats", "unprefixed_malloc_on_supported_platforms"], package = "tikv-jemallocator", optional = true }
+jemallocator-global = { version = "0.4", package = "tikv-jemallocator-global", optional = true }
```
Comment (driftluo) [inline core/run/Cargo.toml]: `jemallocator-global` introduce `cfg-if 0.1` to axon, I think this crate is not necessary

**Comment by driftluo** [comment]
We may need a new make command to start the jemalloc profile with env config

File: Cargo.toml (hunk 1 above)
Comment (KaoImin) [reply Cargo.toml]: Can `profile.release` be controlled by feature?

Comment (driftluo) [reply Cargo.toml]: I think it can‘t, remove debug is enough

**Review by driftluo — APPROVED** [review]

**Pull request merged**